src/CheckIfDead.php

```diff
@@ -412,27 +417,51 @@ public function sanitizeURL( $url ) {
 		if ( isset( $parts['host'] ) ) {
 			// Properly encode the host.  It can't be UTF-8.
 			// See https://en.wikipedia.org/wiki/Internationalized_domain_name.
+			// Hosts are case insensitive, might as well make them lowercase too.
 			if ( function_exists( 'idn_to_ascii' ) ) {
-				$url .= idn_to_ascii( $parts['host'] );
+				$url .= strtolower( idn_to_ascii( $parts['host'] ) );
 			} else {
-				$url .= $parts['host'];
+				$url .= strtolower( $parts['host'] );
 			}
 			if ( isset( $parts['port'] ) ) {
-				$url .= ":" . $parts['port'];
+				// Ports are only needed if not using the defaults for the scheme.
+				// Remove port numbers if they are the default.
+				switch ( $parts['port'] ) {
+					case 80:
+						if ( !isset( $parts['scheme'] ) ||
+							strtolower( $parts['scheme'] ) == "http"
+						) {
+							break;
+						}
+					case 443:
+						if ( isset( $parts['scheme'] ) &&
+							strtolower( $parts['scheme'] ) == "https"
+						) {
+							break;
+						}
+					case 21:
+						if ( isset( $parts['scheme'] ) &&
+							strtolower( $parts['scheme'] ) == "ftp"
+						) {
+							break;
+						}
+					default:
+						$url .= ":" . $parts['port'];
+				}
 			}
 		}
-		// Make sure path, query, and fragment are properly encoded, and not overencoded.
+		// Make sure path, query, and fragment are properly encoded, and not over-encoded.
 		// This avoids possible 400 Bad Response errors.
 		$url .= "/";
 		if ( isset( $parts['path'] ) && strlen( $parts['path'] ) > 1 ) {
 			$url .= implode( '/',
-						array_map( "rawurlencode",
-							explode( '/',
-								substr(
-									urldecode( $parts['path'] ), 1
-								)
-							)
+				array_map( "rawurlencode",
+					explode( '/',
+						substr(
+							urldecode( $parts['path'] ), 1
 						)
```

Comment:
I restored the indenting back to how it's supposed to be here.  No functional change here.